node_client_example/src/client.js

```diff
@@ -35,16 +35,26 @@ var ajaxSettings = {
     // Basic auth params are OK (hardcoding these for the demo nginx proxy)
     // but they only apply to the HTTP requests made by jupuyter-js-services
     // at the moment. https://github.com/jupyter/jupyter-js-services/issues/158
-    user: 'fakeuser',
-    password: 'fakepass',
+    // user: 'fakeuser',
+    // password: 'fakepass',
     // extra headers are OK
-    requestHeaders: {
-        'X-Some-Header': 'some-value'
-    }
+    // requestHeaders: {
+    //     'X-Some-Header': 'some-value'
+    // }
 };
 
+if (process.env.BASE_GATEWAY_USERNAME) {
+    ajaxSettings['user'] = process.env.BASE_GATEWAY_USERNAME
```

Comment:
Since the original example was hardcoded to work with the nginx fakeuser / fakepass, can you add those to the docker-compose.yml as the environment variable settings for process.env.BASE_GATEWAY_USERNAME and process.env.BASE_GATEWAY_PASSWORD so that the example works out of the box?